File: app/src/main/java/com.example.petridelivery.wrappers/base/AuthenticatedRequestInterceptor.java
package com.example.petridelivery.wrappers.base;

import com.example.petridelivery.app.PetriDeliveryApp;
import com.petri.delivery.web.objects.ContDto;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.util.Hashtable;
import java.util.List;

import javax.inject.Inject;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class AuthenticatedRequestInterceptor implements Interceptor {

    private PetriDeliveryApp app;

    public static final String TOKEN = "filter.access.header.username";
    public static final String USERNAME = "filter.access.header.token";

    @Inject
    public AuthenticatedRequestInterceptor(PetriDeliveryApp app) {
        this.app = app;
    }

    @NotNull
    @Override
    public Response intercept(@NotNull Chain chain) throws IOException {
        Request request =  chain.request();

        Hashtable<String, Object> config = app.getConfiguration();
        if(config != null){
            if(((List<String>)config.get("authFilterList")).contains(request.url().uri().toString())){
                ContDto cont = app.getCont();

                final Hashtable<String, String> filters = (Hashtable<String, String>) config.get("filters");
                final String tokenHeader = filters.get(TOKEN);
                final String usernameHeader = filters.get(USERNAME);

                request = request.newBuilder()
                        .addHeader(tokenHeader, cont.getAccesToken())
                        .addHeader(usernameHeader, cont.getNumeDeUtilizator())
                        .build();
            }
        }
        return chain.proceed(request);
    }
}

File: app/src/main/java/com.example.petridelivery.wrappers/base/AuthenticatedRequestInterceptor.kt
package com.example.petridelivery.wrappers.base

import com.example.petridelivery.app.PetriDeliveryApp
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.util.*
import javax.inject.Inject

class AuthenticatedRequestInterceptor @Inject constructor(private val app: PetriDeliveryApp?) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        var request = chain.request()
        val config = app!!.getConfiguration()

        if (config != null) {
            if ((config["authFilterList"] as MutableList<String>).contains(request.url.toUri().toString())) {
                val cont = app.getCont()
                val filters = config["filters"] as Hashtable<String, String>

                val tokenHeader = filters[TOKEN]
                val usernameHeader = filters[USERNAME]

                request = request.newBuilder()
                        .addHeader(tokenHeader!!, cont!!.accesToken)
                        .addHeader(usernameHeader!!, cont.numeDeUtilizator)
                        .build()
            }
        }
        return chain.proceed(request)
    }

    companion object {
        val TOKEN: String? = "filter.access.header.username"
        val USERNAME: String? = "filter.access.header.token"
    }
}